components/raftstore-v2/src/operation/ready/snapshot.rs

```diff
@@ -267,14 +275,25 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
     /// Try to switch snap state to generated. only `Generating` can switch to
     /// `Generated`.
     ///  TODO: make the snap state more clearer, the snapshot must be consumed.
-    pub fn on_snapshot_generated(&self, snap: Box<Snapshot>) -> bool {
+    pub fn on_snapshot_generated(&self, res: GenSnapRes) -> bool {
+        if !res.success {
+            self.cancel_generating_snap(None);
+            return false;
+        }
+        let snap = res.snapshot;
         let mut snap_state = self.snap_state_mut();
         let SnapState::Generating {
             ref canceled,
             ref index,
          } = *snap_state else { return false };
```

Comment:
this condition sometime is mistake.maybe use `!= ` is better.

Reply:
here is else.